### PR TITLE
Upgrade Sunspot to Solr 5

### PR DESCRIPTION
**Pull request by backus**
This pull request upgrades sunspot to use Solr 5 (as requested in #658).  A few notes:

* I removed ruby 1.8.7 from .travis.yml since rsolr has dropped support
* I included PR #662 

I've detailed other significant changes in my commits

### CODE REVIEW

File: sunspot_solr/lib/sunspot/solr/installer/config_installer.rb [hunk 1]
```diff
@@ -13,39 +18,70 @@ def execute(config_path, options = {})
...
+          list = Rake::FileList.new("#{sunspot_config_path}/**/*") do |fl| 
+            fl.include "#{sunspot_config_path}/../solr.xml", "#{sunspot_config_path}/../**/core.properties"
           end
 
-          # Also copy the solr.xml file for multi core support
-          file = File.expand_path('../solr.xml', sunspot_config_path)
-          dest = File.expand_path(File.join(@config_path, ".."), File.basename(file))
-          say("Copying #{file} => #{dest}")
-          FileUtils.cp(file, dest)
+          binding.pry
```
Comment (mois3x) [inline sunspot_solr/lib/sunspot/solr/installer/config_installer.rb]: @backus binding here

Comment (backus) [inline sunspot_solr/lib/sunspot/solr/installer/config_installer.rb]: Woops thanks

**Comment by alainmeier** [comment]
:+1:

**Comment by backus** [comment]
Looks like I dropped a few files while rebasing.  Will update soon

**Comment by barttenbrinke** [comment]
Solr5 no longer accepts time: created_at, which sunspot converts to created_at_d. Probably because it is in fact a date time. Manually changing the XML to created_at_dt works, but this should be addresses.

**Comment by backus** [comment]
Thanks @barttenbrinke. I'll look into that soon

**Comment by Dragsbaek** [comment]
any news on when to expect this to be merged?

**Comment by connorjacobsen** [comment]
:+1:

**Comment by alainmeier** [comment]
Would love to see this excellent PR merged :100:

**Comment by connorjacobsen** [comment]
Would definitely use this :+1:

**Comment by rposborne** [comment]
:+1:

**Comment by killy** [comment]
:+1:

**Comment by alainmeier** [comment]
@njakobsen @joneslee85 can we make this happen? :)

**Comment by devonzuegel** [comment]
Great PR, would merge again :+1:

**Comment by IslamAzab** [comment]
:+1:

**Comment by josephglanville** [comment]
Using this in production, would like to see merged into mainline however.

**Comment by jitensachdeva** [comment]
:+1:  Can we get this merged in mainline

**Comment by njakobsen** [comment]
Thanks for the PR. I finally had a chance to try it out and so far so good.

**Comment by backus** [comment]
Thanks for merging @njakobsen! If there are any issues with code I introduced feel free to @mention me and I'll troubleshoot

**Comment by jitensachdeva** [comment]
thanks @backus . Is there any migration todos for moving from SOLR3 to SOLR5 in terms of changes in schema.xml

**Comment by backus** [comment]
@jitensachdeva 

1. Take a look at my commit here: https://github.com/sunspot/sunspot/commit/4d997c604051569e7f5a1a46deae0f56c0a6456d#diff-3144b2841e6eb43d3fd245dd6591b20c

2. Read [Upgrading from Solr 4.x](http://lucene.apache.org/solr/5_0_0/changes/Changes.html#v5.0.0.upgrading_from_solr_4.x)

2. Here is a diff of the old schema.xml to the new one:

```sh
diff =(curl -s https://raw.githubusercontent.com/sunspot/sunspot/372db4146acfb87ea8363b6a112718d81fb89603/sunspot_solr/solr/solr/conf/schema.xml) =(curl -s https://raw.githubusercontent.com/sunspot/sunspot/master/sunspot_solr/solr/solr/configsets/sunspot/conf/schema.xml)
```

```diff
--- /var/folders/wq/h965ll7s5bqcpzcd58b__pw00000gn/T/zshhz09kF	2015-05-28 02:23:46.000000000 -0700
+++ /var/folders/wq/h965ll7s5bqcpzcd58b__pw00000gn/T/zshBs0sOL	2015-05-28 02:23:46.000000000 -0700
@@ -15,10 +15,10 @@
  See the License for the specific language governing permissions and
  limitations under the License.
 -->
-<!--
+<!--
  This is the Solr schema file. This file should be named "schema.xml" and
  should be in the conf directory under the solr home
- (i.e. ./solr/conf/schema.xml by default)
+ (i.e. ./solr/conf/schema.xml by default)
  or located where the classloader for the Solr webapp can find it.

  This example schema is the recommended starting point for users.
@@ -70,23 +70,17 @@
     <!-- *** This fieldType is used by Sunspot! *** -->
     <fieldType name="boolean" class="solr.BoolField" omitNorms="true"/>
     <!-- *** This fieldType is used by Sunspot! *** -->
-    <fieldType name="date" class="solr.DateField" omitNorms="true"/>
-    <!-- *** This fieldType is used by Sunspot! *** -->
-    <fieldType name="sdouble" class="solr.SortableDoubleField" omitNorms="true"/>
-    <!-- *** This fieldType is used by Sunspot! *** -->
-    <fieldType name="sfloat" class="solr.SortableFloatField" omitNorms="true"/>
-    <!-- *** This fieldType is used by Sunspot! *** -->
-    <fieldType name="sint" class="solr.SortableIntField" omitNorms="true"/>
-    <!-- *** This fieldType is used by Sunspot! *** -->
-    <fieldType name="slong" class="solr.SortableLongField" omitNorms="true"/>
-    <!-- *** This fieldType is used by Sunspot! *** -->
     <fieldType name="tint" class="solr.TrieIntField" omitNorms="true"/>
     <!-- *** This fieldType is used by Sunspot! *** -->
+    <fieldType name="tlong" class="solr.TrieLongField" omitNorms="true"/>
+    <!-- *** This fieldType is used by Sunspot! *** -->
     <fieldType name="tfloat" class="solr.TrieFloatField" omitNorms="true"/>
     <!-- *** This fieldType is used by Sunspot! *** -->
     <fieldType name="tdate" class="solr.TrieDateField"
                omitNorms="true"/>

+    <fieldType name="daterange" class="solr.DateRangeField" omitNorms="true" />
+
     <!-- Special field type for spell correction. Be careful about
          adding filters here, as they apply *before* your values go in
          the spellcheck. For example, the lowercase filter here means
@@ -158,45 +152,45 @@
     <!-- *** This dynamicField is used by Sunspot! *** -->
     <dynamicField name="*_bms" stored="true" type="boolean" multiValued="true" indexed="true"/>
     <!-- *** This dynamicField is used by Sunspot! *** -->
-    <dynamicField name="*_d" stored="false" type="date" multiValued="false" indexed="true"/>
+    <dynamicField name="*_d" stored="false" type="tdate" multiValued="false" indexed="true"/>
     <!-- *** This dynamicField is used by Sunspot! *** -->
-    <dynamicField name="*_dm" stored="false" type="date" multiValued="true" indexed="true"/>
+    <dynamicField name="*_dm" stored="false" type="tdate" multiValued="true" indexed="true"/>
     <!-- *** This dynamicField is used by Sunspot! *** -->
-    <dynamicField name="*_ds" stored="true" type="date" multiValued="false" indexed="true"/>
+    <dynamicField name="*_ds" stored="true" type="tdate" multiValued="false" indexed="true"/>
     <!-- *** This dynamicField is used by Sunspot! *** -->
-    <dynamicField name="*_dms" stored="true" type="date" multiValued="true" indexed="true"/>
+    <dynamicField name="*_dms" stored="true" type="tdate" multiValued="true" indexed="true"/>
     <!-- *** This dynamicField is used by Sunspot! *** -->
-    <dynamicField name="*_e" stored="false" type="sdouble" multiValued="false" indexed="true"/>
+    <dynamicField name="*_e" stored="false" type="tdouble" multiValued="false" indexed="true"/>
     <!-- *** This dynamicField is used by Sunspot! *** -->
-    <dynamicField name="*_em" stored="false" type="sdouble" multiValued="true" indexed="true"/>
+    <dynamicField name="*_em" stored="false" type="tdouble" multiValued="true" indexed="true"/>
     <!-- *** This dynamicField is used by Sunspot! *** -->
-    <dynamicField name="*_es" stored="true" type="sdouble" multiValued="false" indexed="true"/>
+    <dynamicField name="*_es" stored="true" type="tdouble" multiValued="false" indexed="true"/>
     <!-- *** This dynamicField is used by Sunspot! *** -->
-    <dynamicField name="*_ems" stored="true" type="sdouble" multiValued="true" indexed="true"/>
+    <dynamicField name="*_ems" stored="true" type="tdouble" multiValued="true" indexed="true"/>
     <!-- *** This dynamicField is used by Sunspot! *** -->
-    <dynamicField name="*_f" stored="false" type="sfloat" multiValued="false" indexed="true"/>
+    <dynamicField name="*_f" stored="false" type="tfloat" multiValued="false" indexed="true"/>
     <!-- *** This dynamicField is used by Sunspot! *** -->
-    <dynamicField name="*_fm" stored="false" type="sfloat" multiValued="true" indexed="true"/>
+    <dynamicField name="*_fm" stored="false" type="tfloat" multiValued="true" indexed="true"/>
     <!-- *** This dynamicField is used by Sunspot! *** -->
-    <dynamicField name="*_fs" stored="true" type="sfloat" multiValued="false" indexed="true"/>
+    <dynamicField name="*_fs" stored="true" type="tfloat" multiValued="false" indexed="true"/>
     <!-- *** This dynamicField is used by Sunspot! *** -->
-    <dynamicField name="*_fms" stored="true" type="sfloat" multiValued="true" indexed="true"/>
+    <dynamicField name="*_fms" stored="true" type="tfloat" multiValued="true" indexed="true"/>
     <!-- *** This dynamicField is used by Sunspot! *** -->
-    <dynamicField name="*_i" stored="false" type="sint" multiValued="false" indexed="true"/>
+    <dynamicField name="*_i" stored="false" type="tint" multiValued="false" indexed="true"/>
     <!-- *** This dynamicField is used by Sunspot! *** -->
-    <dynamicField name="*_im" stored="false" type="sint" multiValued="true" indexed="true"/>
+    <dynamicField name="*_im" stored="false" type="tint" multiValued="true" indexed="true"/>
     <!-- *** This dynamicField is used by Sunspot! *** -->
-    <dynamicField name="*_is" stored="true" type="sint" multiValued="false" indexed="true"/>
+    <dynamicField name="*_is" stored="true" type="tint" multiValued="false" indexed="true"/>
     <!-- *** This dynamicField is used by Sunspot! *** -->
-    <dynamicField name="*_ims" stored="true" type="sint" multiValued="true" indexed="true"/>
+    <dynamicField name="*_ims" stored="true" type="tint" multiValued="true" indexed="true"/>
     <!-- *** This dynamicField is used by Sunspot! *** -->
-    <dynamicField name="*_l" stored="false" type="slong" multiValued="false" indexed="true"/>
+    <dynamicField name="*_l" stored="false" type="tlong" multiValued="false" indexed="true"/>
     <!-- *** This dynamicField is used by Sunspot! *** -->
-    <dynamicField name="*_lm" stored="false" type="slong" multiValued="true" indexed="true"/>
+    <dynamicField name="*_lm" stored="false" type="tlong" multiValued="true" indexed="true"/>
     <!-- *** This dynamicField is used by Sunspot! *** -->
-    <dynamicField name="*_ls" stored="true" type="slong" multiValued="false" indexed="true"/>
+    <dynamicField name="*_ls" stored="true" type="tlong" multiValued="false" indexed="true"/>
     <!-- *** This dynamicField is used by Sunspot! *** -->
-    <dynamicField name="*_lms" stored="true" type="slong" multiValued="true" indexed="true"/>
+    <dynamicField name="*_lms" stored="true" type="tlong" multiValued="true" indexed="true"/>
     <!-- *** This dynamicField is used by Sunspot! *** -->
     <dynamicField name="*_s" stored="false" type="string" multiValued="false" indexed="true"/>
     <!-- *** This dynamicField is used by Sunspot! *** -->
@@ -241,6 +235,14 @@
     <dynamicField name="*_ets" stored="true" termVectors="true" type="tdouble" multiValued="false" indexed="true"/>
     <!-- *** This dynamicField is used by Sunspot! *** -->
     <dynamicField name="*_etms" stored="true" termVectors="true" type="tdouble" multiValued="true" indexed="true"/>
+    <!-- *** This dynamicField is used by Sunspot! *** -->
+    <dynamicField name="*_dr" stored="false" type="daterange" multiValued="false" indexed="true" />
+    <!-- *** This dynamicField is used by Sunspot! *** -->
+    <dynamicField name="*_drm" stored="false" type="daterange" multiValued="true" indexed="true" />
+    <!-- *** This dynamicField is used by Sunspot! *** -->
+    <dynamicField name="*_drs" stored="true" type="daterange" multiValued="false" indexed="true" />
+    <!-- *** This dynamicField is used by Sunspot! *** -->
+    <dynamicField name="*_drms" stored="true" type="daterange" multiValued="true" indexed="true" />

     <!-- Type used to index the lat and lon components for the "location" FieldType -->
     <dynamicField name="*_coordinate"  type="tdouble" indexed="true"  stored="false" multiValued="false"/>
@@ -251,10 +253,11 @@
     <dynamicField name="*_lls" stored="true" type="location" multiValued="false" indexed="true"/>
     <dynamicField name="*_llms" stored="true" type="location" multiValued="true" indexed="true"/>
     <field name="textSpell" stored="false" type="textSpell" multiValued="true" indexed="true"/>
+
     <!-- required by Solr 4 -->
     <field name="_version_" type="string" indexed="true" stored="true" multiValued="false" />
   </fields>
-
+
   <!-- Field to use to determine and enforce document uniqueness.
       Unless this field is marked with required="false", it will be a required field
    -->
```

**Comment by jitensachdeva** [comment]
@backus thanks.. this is very helpful..

**Pull request merged**